inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/JsonArrayInsertFunction.java

```diff
@@ -0,0 +1,124 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.sdk.transform.process.function;
+
+import org.apache.inlong.sdk.transform.decode.SourceData;
+import org.apache.inlong.sdk.transform.process.Context;
+import org.apache.inlong.sdk.transform.process.operator.OperatorTools;
+import org.apache.inlong.sdk.transform.process.parser.ValueParser;
+
+import com.alibaba.fastjson.JSON;
+import com.alibaba.fastjson.JSONArray;
+import com.alibaba.fastjson.JSONObject;
+import com.alibaba.fastjson.JSONPath;
+import net.sf.jsqlparser.expression.Expression;
+import net.sf.jsqlparser.expression.Function;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * JsonArrayInsertFunction   ->  JSON_ARRAY_INSERT(json_doc, path, val[, path, val] ...)
+ * description:
+ * - return NULL if any argument is NULL;
+ * - return the document inserted into the array.
+ * Note: If multiple groups of parameters are passed in, the parameter subscripts of the latter groups
+ * need to be based on the document subscripts after the previous group of parameters are updated.
+ */
+@TransformFunction(names = {"json_array_insert"})
+public class JsonArrayInsertFunction implements ValueParser {
+
+    private ValueParser jsonDocParser;
+    private List<ValueParser> pathValuePairsParser;
+
+    public JsonArrayInsertFunction(Function expr) {
+        List<Expression> expressions = expr.getParameters().getExpressions();
+        jsonDocParser = OperatorTools.buildParser(expressions.get(0));
+        pathValuePairsParser = new ArrayList<>();
+        for (int i = 1; i < expressions.size(); i++) {
+            pathValuePairsParser.add(OperatorTools.buildParser(expressions.get(i)));
+        }
+    }
+
+    @Override
+    public Object parse(SourceData sourceData, int rowIndex, Context context) {
+        Object jsonDocObj = jsonDocParser.parse(sourceData, rowIndex, context);
+        if (jsonDocObj == null) {
+            return null;
+        }
+        ArrayList<Object> pathValuePairs = new ArrayList<>();
+        for (ValueParser valueParser : pathValuePairsParser) {
+            pathValuePairs.add(valueParser.parse(sourceData, rowIndex, context));
+        }
+        return jsonArrayInsert(jsonDocObj.toString(), pathValuePairs);
+    }
+
+    public static String jsonArrayInsert(String jsonDoc, ArrayList<Object> pathValuePairs) {
```

Comment:
private? doesn’t need static I think

Reply:
Thk! I think it makes sense and have modified it.